evap/staff/importers.py

```diff
@@ -97,28 +98,99 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
         assert not self.errors
         # This is safe because the user's email address is checked before in the importer (see #953)
         responsible_dbobj = UserProfile.objects.get(email=self.responsible_email)
-        course = Course(
-            name_de=self.name_de,
-            name_en=self.name_en,
-            type=self.course_type,
+        if self.existing_course:
+            self.existing_course.degrees.add(*self.degrees)
+        else:
+            course = Course(
+                name_de=self.name_de,
+                name_en=self.name_en,
+                type=self.course_type,
+                semester=semester,
+            )
+            course.save()
+            course.responsibles.set([responsible_dbobj])
+            course.degrees.set(self.degrees)
+            evaluation = Evaluation(
+                vote_start_datetime=vote_start_datetime,
+                vote_end_date=vote_end_date,
+                course=course,
+                wait_for_grade_upload_before_publishing=self.is_graded,
+            )
+            evaluation.save()
+            evaluation.contributions.create(
+                evaluation=evaluation,
+                contributor=responsible_dbobj,
+                role=Contribution.Role.EDITOR,
+                textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+            )
+
+    def check_existing_course(self, semester, importer):
+        name_collision = self.check_name_collision(semester)
+        if not name_collision:
+            return False
+
+        self.find_and_set_existing_course(semester, importer)
+        if self.existing_course is None:
+            if name_collision == "both":
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} ({}) does already exist in this semester.").format(self.name_en, self.name_de)
+                )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} does already exist in this semester.").format(getattr(self, name_collision))
+                )
+
+        return self.existing_course is not None
+
+    def check_name_collision(self, semester):
+        name_collision = None
+        if Course.objects.filter(semester=semester, name_en=self.name_en).exists():
+            name_collision = "name_en"
```

Comment:
This should probably be a [Flag](https://docs.python.org/3/library/enum.html#flag). I'd say this whole method can then be inlined.

Reply:
I was thinking something like:
```python3
Course.objects.filter(semester=semester, name_en=self.name_en).exists() and NameCollision.NAME_EN
| Course.objects.filter(semester=semester, name_en=self.name_de).exists() and NameCollision.NAME_DE
```

@niklasmohrin what do you think about abusing the "X and Y returns Y if X is truthy" thing here?

Reply:
I hope https://github.com/e-valuation/EvaP/pull/1683/commits/54a459ef4111a5462e00b490f950cc3c448e2c30 does what was expected

Reply:
Sorry, this didn't work out in the one-line fashion we were aiming at due to the type-error reasons you listed; see the new comment I made below :sweat_smile: